### PR TITLE
Adding unit test for 'uuf-core' component

### PR DESCRIPTION
**Pull request by this**
This PR adds more unit tests for 'uuf-core' component module.
Also upgraded,
- TestNG version to `6.9.10`
- Mockito version to `2.4.2`

### CODE REVIEW

**Comment by this** [comment]
Merge this PR after merging #139.

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java [hunk 1]
```diff
@@ -58,22 +76,178 @@ private static HttpRequest createRequest(String contextPath, String uriWithoutCo
...
+                                      emptySet(), null);
+        Page rootPage = createPage("/x/y/", null);
+        Component rootComponent = new Component("root", null, Component.ROOT_COMPONENT_CONTEXT_PATH,
+                                                ImmutableSortedSet.of(rootPage), emptySet(), emptySet(),
+                                                singleton(cmp), null);
+        App app = new App(null, "/test", ImmutableSet.of(cmp, rootComponent), emptySet(), createConfiguration(), null,
+                          null);
+        PageRedirectException pre;
+
+        pre = Assert.expectThrows(PageRedirectException.class,
```
Comment (Shan1024) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: We can merge these two lines to single line, right?

**Comment by Shan1024** [comment]
I get the following exception when building the product.

```
2017-01-16 12:20:16 INFO  AppCreator:263 - Deployed REST API 'org.wso2.carbon.uuf.internal.deployment.RestApiDeploymentTest$SampleMicroservice' for component 'org.wso2.carbon.uuf.sample.sample-app.feature' with context path '/sample-app/root/apis/sample-api'.
Tests run: 161, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.296 sec <<< FAILURE! - in TestSuite
testLookup(org.wso2.carbon.uuf.core.LookupTest)  Time elapsed: 0.009 sec  <<< FAILURE!
java.util.NoSuchElementException: No value present
        at java.util.Optional.get(Optional.java:135)
        at org.wso2.carbon.uuf.core.LookupTest.testLookup(LookupTest.java:59)
```

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java (hunk 1 above)
Comment (this) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: `pre` variable is again assigned in line 186. IMO this way looks better.

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/ThemeTest.java [hunk 2]
```diff
@@ -1,27 +1,27 @@
 /*
- *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Kishanthan) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/ThemeTest.java]: We don't need to change the year

Comment (this) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/ThemeTest.java]: fixed in commit ecaf59bb73cb1d75b9aa0998a6f68726037841af

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/API.java [hunk 3]
```diff
@@ -128,8 +128,8 @@ public static void callMicroService() {
     }
 
     public static void sendError(int status, String message) {
-        if ((status < 100) || (status > 599)) {
-            throw new IllegalArgumentException("HTTP status code must be between 100 (inclusive) and 600 (exclusive).");
+        if ((status < 100) || (status > 999)) {
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/API.java]: fixed in commit db61108678c34f5c094ddd444ed6c7524bd1c708

**Pull request merged**